### PR TITLE
Add Layout Conditions

### PR DESCRIPTION
**Pull request by MatyasKriz**
EBNF as of making this parser:
**CONDITION** := '[' EXPRESSION ']';
**EXPRESSION** := TERM [ or TERM ];
**TERM** := COMPARISON [ and COMPARISON ];
**COMPARISON** := FACTOR [ ( == | != | ':gt' | ':gte' | ':lt' | ':lte' ) FACTOR ];
**FACTOR** := [ '!' ] ( '(' EXPRESSION ')' | IDENTIFIER | FLOAT_NUMBER );
**FLOAT_NUMBER** := { same as Swift's `Float` }
**IDENTIFIER** := { all cases in the ConditionStatement enum };

Also in this PR:
- some changes I don't even know about,
- added `:gte` and `:lte` aliases to constraints to replace misleading `:gt` and `:lt`,
- some generator refactoring.

### CODE REVIEW

File: Example/Source/Component/SelectionCell/SelectionCell.swift
Comment (matoushybl) [inline Example/Source/Component/SelectionCell/SelectionCell.swift:33]: Why? Why is Reactant downgraded back to 1.1.0?

File: Example/Source/Component/Playground/ExampleRootView.ui.xml
Comment (TadeasKriz) [inline Example/Source/Component/Playground/ExampleRootView.ui.xml:39]: `... or width.lessThan?(100)`

Comment (MatyasKriz) [reply Example/Source/Component/Playground/ExampleRootView.ui.xml:39]: If we're making the parser into function-parsing grade, we should decide on that now otherwise it will get uglier and uglier.

File: Sources/Generator/UIGenerator.swift
Comment (matoushybl) [inline Sources/Generator/UIGenerator.swift]: What do you think about a method that would collect names of the children? Very similar code is used in `generateChildProperties `.

File: Sources/Generator/UIGenerator.swift
Comment (matoushybl) [inline Sources/Generator/UIGenerator.swift:144]: Can this happen in any other case than that the generator is broken?

File: Sources/Generator/UIGenerator.swift
Comment (matoushybl) [inline Sources/Generator/UIGenerator.swift]: What about this dead code?

File: Sources/Generator/UIGenerator.swift
Comment (matoushybl) [inline Sources/Generator/UIGenerator.swift]: What did these dead lines do?

File: Sources/Generator/UIGenerator.swift
Comment (matoushybl) [inline Sources/Generator/UIGenerator.swift]: Generator should support exceptions now so the `try!` should not be needed.

File: Sources/Tokenizer/Layout/Constraints/Constraint.swift
Comment (matoushybl) [inline Sources/Tokenizer/Layout/Constraints/Constraint.swift]: Why not return `swiftRepresentation` without switching?

**Review by matoushybl — APPROVED** [review]
Nice work!

File: Sources/Tokenizer/Layout/Constraints/Constraint.swift
Comment (MatyasKriz) [reply Sources/Tokenizer/Layout/Constraints/Constraint.swift]: It's premature change for future maintainer when we wants to add new unary operator. I guess we can have a dumb repeater for now.

File: Sources/Generator/UIGenerator.swift
Comment (MatyasKriz) [reply Sources/Generator/UIGenerator.swift]: Wait, I thought I removed all of these pesky triers.

File: Example/Source/Component/SelectionCell/SelectionCell.swift
Comment (TadeasKriz) [reply Example/Source/Component/SelectionCell/SelectionCell.swift:33]: Was the `ControlBase` released though?

Comment (MatyasKriz) [reply Example/Source/Component/SelectionCell/SelectionCell.swift:33]: I have no memory of changing this.

File: Sources/Generator/UIGenerator.swift
Comment (MatyasKriz) [reply Sources/Generator/UIGenerator.swift]: I'll be honest, it's quite a lot of time since we changed this file with @TadeasKriz and it's mainly his work in this file.

Comment (matoushybl) [reply Sources/Generator/UIGenerator.swift]: No problem, as long as everything still works. I understand that this part of the project is a mess.

File: Sources/Generator/UIGenerator.swift
Comment (MatyasKriz) [reply Sources/Generator/UIGenerator.swift]: I think the code might have been blatantly copied and slightly modified, that's why they're so similar.

File: Sources/Generator/UIGenerator.swift
Comment (matoushybl) [inline Sources/Generator/UIGenerator.swift]: Does this if make sense since there will always be at least `target`?

Comment (MatyasKriz) [reply Sources/Generator/UIGenerator.swift]: Nice catch.

**Pull request merged**
